src/extension/PostProblems.ts

```diff
@@ -0,0 +1,332 @@
+      )
+      return
+    }
+
+    const decoder = new TextDecoder()
+    const content = decoder.decode(data)
+    const resource_error = this.getResourceError(content)
+
+    this.createAndPostDiagnostics(resource_error, confFile, ts)
+  }
+
+  /** private methods: */
+
+  /** returns a pattern to only watch the files that have problems */
+  private getPatternForFilesToWatch(): string {
+    const wsFolder = vscode.workspace.workspaceFolders?.[0].uri
+    if (!wsFolder) {
+      return '**/'
+    }
+    const uriPatterns: string[] = []
+    this.diagnosticCollection.forEach(collection => {
+      collection.forEach(uri => {
+        const relativePath = uri.path.split(wsFolder.path + '/').join()
+        uriPatterns.push(relativePath)
+      })
+    })
+    const stringPattern = uriPatterns.join(',')
+    return `**/*{${stringPattern}}`
+  }
+
+  /**
+   * when user edits the file where problem originated, clear related diagnostics
+   */
+  private async deleteOnEdit(): Promise<void> {
+    // counts the diagnostic collections left
+    let diagnosticCollectionsLength: number = this.diagnosticCollection.length
+    const folder = workspace.workspaceFolders?.[0]
+    const pattern = this.getPatternForFilesToWatch()
+    if (folder) {
+      const watcher = vscode.workspace.createFileSystemWatcher(
+        new vscode.RelativePattern(folder, pattern),
+      )
+      watcher.onDidChange(uri => {
+        this.diagnosticCollection.forEach(collection => {
+          collection.forEach(diagnosticUri => {
+            if (uri.path === diagnosticUri.path) {
+              collection.clear()
+              diagnosticCollectionsLength -= 1
+            }
+            // if no more diagnostics left - stop watching
+            if (diagnosticCollectionsLength === 0) {
+              this.diagnosticCollection = []
+              watcher.dispose()
+            }
+          })
+        })
+      })
+    }
+  }
+
+  /**
+   * Creates the diagnostics and posts them.
+   * A diagnostic will link to the file the error originated from if a path to such file exists in the message.
+   * If not - the diagnostic will link to the .conf.log file of the current run.
+   * @param resource_error json content of resource_error.json
+   * @param confFile a relative path to this run .conf file
+   * @param ts the timestamp of the current run
+   */
+  private async createAndPostDiagnostics(
+    resource_error: ResourceError,
+    confFile: string,
+    ts: number,
+  ): Promise<void> {
+    /**
+     * regex to find a file path in a string. example:
+     * string: "BankLesson/Bank.sol:22:5: ParserError: Expected ';' but got 'function'"
+     * pathRegex will find: BankLesson/Bank.sol
+     * (changes to include relative paths)
+     */
+    const pathRegex = /([a-z0-9_\-\\/.]+)\.([a-z0-9]+)/i
+    const locationRegex = /((:\d+:\d+:)|(:\d+:\d+))/g
+
+    const diagnosticMap: Map<string, vscode.Diagnostic[]> = new Map()
```

Comment:
The reason I used `string` instead of `Uri` is that when I fill the diagnostic map, I want every diagnostic object that is linked to the same uri to be in the same diagnostic list, for example if I have:
`diagnostic1` from `some_file`
`diagnostic2` from `some_file`
`diagnostic3` from `other_file`
`diagnostic4` from `other_file`
I want them to be mapped as follows:
`<some_file, [diagnostic1, diagnostic2]>, <other_file, [diagnostic3, diagnostic4]>`

now to get a value efficiently from the map, we need to use inner compare, in my code it looks like this:
`const diagnostics: Diagnostic[] = diagnosticMap.get(path) || []` (line 203)

the `path` here is the new path we got from this iteration. if we wanted to use a `Uri,` we needed to create a new uri object from this file but them the inner compare will always show us `false` as these are 2 different object.
I can solve it by using `foreach` and then compare: `new_uri.path === old_uri.path` but it seems like a lot of unnecessary effort.